tests/test_arcaflow_plugin_pcp.py

```diff
@@ -146,6 +146,46 @@ def test_functional_flat(self):
         self.assertEqual("success", output_id)
         self.assertIsInstance(output_data.pcp_output, list)
 
+    def test_functional_user_pmlogger_conf(self):
+        f = open("tests/pmlogger.conf", "r")
+        pmlogger_conf = f.read()
+        input = pcp_plugin.PcpInputParams(
+            pmlogger_interval=1.0,
+            pmlogger_metrics="kernel.all.cpu.user mem.util.used",
+            timeout=3,
+            pmlogger_conf=pmlogger_conf,
+        )
```

Comment:
It's slightly unsatisfying to open a file, read it, and then hand the contents to the plugin as a string so that it can write them into another file to hand to PCP to open and read.

I presume that it is useful for the plugin to accept the configuration as a string, but it seems likely that it would also be good if the plugin could accept it as a file which it could hand directly to PCP, without the rest of the dance being required.

Reply:
What you are proposing would be a change to the SDKs, not the plugin. AFAIK right now the only file input methods are being provided by the workflow engine, and the plugin SDKs only allow for JSON/YAML as inputs. The test configuration files are very large and simpler to maintain as separate files for the test, so we need to concatenate that file into the JSON that we are passing to the plugin.

Reply:
If the input to the plugin were a string, the SDK and engine would pass its value to the plugin, right?  And, then it would be up to the plugin to decide whether the string was the name of a file or not...right?  What am I missing?

For instance, it seems to me that this PR is adding an input to the plugin called `pmrep_conf` and that this is an optional parameter.  What if we added another input parameter called (e.g.) `pmrep_conf_path`, which is a string containing a path to a file containing a configuration?  We could set its default value to `/etc/pcp/pmrep` and then use that [here](https://github.com/arcalot/arcaflow-plugin-pcp/blob/5cb51b0854f94e2e681162698fd5d9d8fccb79a1/arcaflow_plugin_pcp/pcp_plugin.py#L102) if `pmrep_conf` is empty.  I think that would give you everything that you have now _plus_ the ability to pass in a configuration as a file instead of as a ginormous string that has to be repeatedly read and written.

Reply:
The plugin in normal conditions runs as a container, so a path provided will be relative to the container namespace. That means that the user must also run the container with a file system mapping to the the host source location of the file being referenced. That isn't terribly complicated when running the plugin standalone, but would be pretty complex within a workflow (though in that case we can use the engine's method of file passing).

All in all, probably too much to consider for this PR.

Reply:
> The plugin in normal conditions runs as a container, so a path provided will be relative to the container namespace.

That does create a bit of a wrinkle.  I know that there is a mechanism for providing bind mounts for containers; I don't recall the details of how a workflow would access it.